examples/gno.land/p/demo/simpledao/membstore.gno

```diff
@@ -0,0 +1,181 @@
+package simpledao
+
+import (
+	"errors"
+
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/gov/dao"
+)
+
+var (
+	errAlreadyMember        = errors.New("address is already a member")
+	errMissingMember        = errors.New("address is not a member")
+	errInvalidAddressUpdate = errors.New("invalid address update")
+)
+
+type MembStoreOption func(*MembStore)
+
+// WithInitialMembers initializes the simpledao member store
+// with an initial member list
+func WithInitialMembers(members []dao.Member) MembStoreOption {
+	return func(store *MembStore) {
+		for _, m := range members {
+			store.members.Set(m.Address.String(), m)
+
+			store.totalVotingPower += m.VotingPower
+		}
+
+		store.majorityPower = (2 * store.totalVotingPower) / 3
+	}
+}
+
+// MembStore implements the dao.MembStore abstraction
+type MembStore struct {
+	members *avl.Tree // std.Address -> dao.Member
+
+	totalVotingPower uint64 // cached value for quick lookups
+	majorityPower    uint64 // cached value for quick lookups
+}
+
+// NewMembStore creates a new simpledao member store
+func NewMembStore(opts ...MembStoreOption) *MembStore {
+	m := &MembStore{
+		members:          avl.NewTree(),
+		totalVotingPower: 0,
+		majorityPower:    0,
+	}
+
+	// Apply the options
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	return m
+}
+
+func (m *MembStore) AddMember(member dao.Member) error {
+	if !isCallerGOVDAO() {
+		return errNotGovDAO
+	}
+
+	// Check if the member exists
+	if m.IsMember(member.Address) {
+		return errAlreadyMember
+	}
+
+	// Add the member
+	m.members.Set(member.Address.String(), member)
+
+	// Update the total voting power
+	m.totalVotingPower += member.VotingPower
+	m.majorityPower = (2 * m.totalVotingPower) / 3
+
+	return nil
+}
+
+func (m *MembStore) UpdateMember(address std.Address, member dao.Member) error {
+	if !isCallerGOVDAO() {
+		return errNotGovDAO
+	}
+
+	// Check if the member exists
+	memberRaw, exists := m.members.Get(address.String())
+	if !exists {
+		return errMissingMember
+	}
+
+	// Check if this is a removal request
+	if member.VotingPower == 0 {
+		m.members.Remove(address.String())
+
+		return nil
+	}
```

Comment:
cache values are not updated here. Several corner cases on this method need to update these cache values.

Reply:
This is a good catch 💯 

I've fixed it up, and made sure we cover it in the tests:
[d320493](https://github.com/gnolang/gno/pull/2581/commits/d3204930c11c7db5a20ef8bb0cc1bdc14d19f68b)